hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -281,9 +281,10 @@ && equal( join.getKey(), manyToOne ) ) {
 				if ( CollectionPart.Nature.fromNameExact( navigableRole.getParent().getLocalName() ) != null ) {
 					// * the to-one's parent is directly a collection element or index
 					// * therefore, its parent-parent should be the collection itself
-					final String pluralAttributeName = StringHelper.unqualify( navigableRole.getParent().getParent().getLocalName() );
-					final PluralAttributeMapping pluralAttribute = (PluralAttributeMapping) declaringEntityPersister.findAttributeMapping( pluralAttributeName );
+					final PluralAttributeMapping pluralAttribute = (PluralAttributeMapping) declaringEntityPersister.findByPath(
+							navigableRole.getParent().getParent().getLocalName() );
```

Comment:
I think the change in `findByPath` becomes unnecessary when you use this instead:

```suggestion
							navigableRole.getParent().getParent().getFullPath().substring( declaringEntityPersister.getNavigableRole().getFullPath().length() + 1 ) );
```

I guess that the problem you are facing is that the navigable role of a collection persister contains the fully qualified class name of the entity? This should fix it.

Reply:
it was one of my possible solutions but I did not like it :P
Do you think it is better than the change to `findByPath()`?

Reply:
The change in `findByPath` is not correct, sorry :|
If you think about it, it's easily possible to construct an example that will produce wrong results due to that change in `findByPath`